studio-docs/source/embed-explorer.mdx

```diff
@@ -0,0 +1,671 @@
+
+> If you use this method, set the `sendRequestsFrom` query parameter to `embed`.
+>
+> Also make sure you've set your public variant's endpoint URL (and its subscription endpoint, if any) in Apollo Studio.
+
+After you add the Explorer's origin to your endpoint's CORS configuration, you're all set! You can now embed the `iframe` in any webpage and set its dimensions as you like. Also check out [additional supported options](#options).
+
+### Route requests through the parent page (advanced)
+
+Some endpoints _can't_ receive requests from the Explorer's origin due to firewall rules or organizational policies. In these cases, you can use the the `postMessage` API to route Explorer requests through your embedding webpage's domain.
+
+> If you use this method, set the `sendRequestsFrom` query parameter to `parent`.
+
+If you need to use this advanced method, proceed to the additional setup in [Using `postMessage`](#using-postmessage-advanced).
+
+## Options
+
+When you embed the Explorer, you set options for it by including query parameters in the `src` URL of the `iframe`, like so:
+
+```jsx
+<iframe src="https://explorer.embed.apollographql.com/?graphRef=my-graph-id@my-variant&sendRequestsFrom=embed" />
+```
+
+The following options are supported:
+
+<table class="field-table api-ref">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+<tr class="required">
+<td>
+
+##### `graphRef`
+
+`string`
+
+</td>
+<td>
+
+**Required.** The graph ref for the [public variant](./org/graphs/#public-variants) you want to use the embedded Explorer with. Has the format `graph-id@variant-name`.
+
+The Explorer fetches this variant's schema from Apollo Studio to populate its Documentation panel and enable code completion.
+
+If you omit this option, the Explorer _does_ still load, but it doesn't know which schema to fetch. This prevents the Explorer from providing critical features like documentation and code completion.
+
+</td>
+</tr>
+
+<tr class="required">
+<td>
+
+##### `sendRequestsFrom`
+
+`"embed" | "parent"`
+
+</td>
+<td>
+
+**Required.** If `parent`, the embedded Explorer uses the `postMessage` API to notify the parent page when a user initiates a GraphQL operation. The parent page is then responsible for transmitting the operation to your GraphQL server.
+
+If `embed`, the embedded Explorer transmits operations directly to your GraphQL server from the origin `https://explorer.embed.apollographql.com`.
+
+See [Handling CORS](#handling-cors).
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+##### `document`
+
+`string`
+
+</td>
+<td>
+
+A URI-encoded operation to populate in the Explorer's editor on load.
+
+If you omit this, the Explorer initially loads an example query based on your schema.
+
+If [`shouldPersistState`](#shouldpersiststate) is `true`, the editor is _also_ populated with user-defined operations from `localStorage`.
```

Comment:
```suggestion
If [`shouldPersistState`](#shouldpersiststate) is `true`, and you provide a document, the editor will open a new tab with this document on load alongside any tabs the user had open from `localStorage`.
```

Reply:
^ this is also true for `variables` and `headers`. I wanted to make it clear that we just open the state you provide in the url (`document`, `variables`, `headers` in a new tab on load, regardless of the local storage situation)